### PR TITLE
feat(server): run microservices in worker thread

### PR DESCRIPTION
**Pull request by zackpollard**
POC for bringing microservices into the server instance by using Node.JS worker threads.

### CODE REVIEW

File: server/src/main.ts [hunk 1]
```diff
@@ -86,15 +73,28 @@ async function bootstrapImmichAdmin() {
...
+      process.exit(exitCode);
+    }
+  });
+}
+
 function bootstrap() {
   switch (immichApp) {
     case 'immich': {
       process.title = 'immich_server';
+      if (process.env.INTERNAL_MICROSERVICES === 'true') {
```
Comment (bo0tzz) [inline server/src/main.ts:90]: Do we eventually want to have an env var for each worker, or rather something like `IMMICH_ENABLED_WORKERS=api,web,microservices`?

**Review by bo0tzz — APPROVED** [review]
🚀

File: server/src/repositories/logger.repository.ts [hunk 2]
```diff
@@ -10,6 +10,12 @@ export class LoggerRepository extends ImmichLogger implements ILoggerRepository
     super(LoggerRepository.name);
   }
 
+  static appName?: string = undefined;
```
Comment (danieldietzler) [inline server/src/repositories/logger.repository.ts:13]: Shouldn't this be private? Or is that implicit?

**Review by danieldietzler — APPROVED** [review]

File: server/src/main.ts (hunk 1 above)
Comment (zackpollard) [reply server/src/main.ts:90]: We can figure this out later, but a pattern like that might make sense

**Pull request merged**